library/src/main/java/com/mapzen/android/MapzenMap.java

```diff
@@ -676,4 +708,26 @@ public void drawTransitRouteLine(@NotNull List<LngLat> points, @Nullable List<Ln
   public void clearTransitRouteLine() {
     overlayManager.clearTransitRouteLine();
   }
+
+  /**
+   * By default all {@link MapData} objects are removed from the map when it is destroyed. To
+   * persist this data such as in the case of orientation changes, use this method.
+   * @param persistOnRecreation persist {@link MapData} across orientation changes
+   */
+  public void setPersistMapData(boolean persistOnRecreation) {
```

Comment:
Also I'm starting to think the default here should be that the data is persisted because now the sample app feels broken when you rotate it and the map state resets each time.